src/http_client_async.zig

```diff
@@ -2621,6 +2621,8 @@ pub fn onData(this: *HTTPClient, comptime is_ssl: bool, incoming_data: []const u
             if (this.state.content_encoding_i < response.headers.len and !this.state.did_set_content_encoding) {
                 // if it compressed with this header, it is no longer because we will decompress it
                 var mutable_headers = std.ArrayListUnmanaged(picohttp.Header){ .items = response.headers, .capacity = response.headers.len };
+                // we remove the content encoding header
+                _ = mutable_headers.swapRemove(this.state.content_encoding_i);
```

Comment:
This will change the order that the headers are passed to the server, which is technically observable